mri/operators/linear/wavelet.py

```diff
@@ -57,60 +67,89 @@ def get_coeff(self):
     def set_coeff(self, coeffs):
         self.transform.analysis_data = coeffs
 
+    def _op(self, data):
+        if isinstance(data, np.ndarray):
+            data = pysap.Image(data=data)
+        self.transform.data = data
+        self.transform.analysis()
+        coeffs, coeffs_shape = flatten(self.transform.analysis_data)
+        return coeffs, coeffs_shape
+
     def op(self, data):
         """ Define the wavelet operator.
-
         This method returns the input data convolved with the wavelet filter.
-
         Parameters
         ----------
         data: ndarray or Image
             input 2D data array.
-
         Returns
         -------
         coeffs: ndarray
             the wavelet coefficients.
         """
-        if isinstance(data, np.ndarray):
-            data = pysap.Image(data=data)
-        self.transform.data = data
-        self.transform.analysis()
-        coeffs, self.coeffs_shape = flatten(self.transform.analysis_data)
+        if self.n_coils > 1:
+            coeffs, self.coeffs_shape = zip(*Parallel(n_jobs=self.n_jobs,
+                                                      backend=self.backend,
+                                                      verbose=self.verbose)(
+                delayed(self._op)
+                (data[i])
+                for i in np.arange(self.n_coils)))
```

Comment:
This multi-line statement is difficult to read.
Why not:
```python
coeffs, self.coeffs_shape = zip(
	*Parallel(n_jobs=self.n_jobs, backend=self.backend, verbose=self.verbose)(
                delayed(self._op)(data[i]) for i in np.arange(self.n_coils)
	)
)
```
If too big then:
```python
coeffs, self.coeffs_shape = zip(
	*Parallel(
		n_jobs=self.n_jobs, 
		backend=self.backend, 
		verbose=self.verbose
	)(
      	delayed(self._op)(data[i]) 
		for i in np.arange(self.n_coils)
	)
)
```
or you could even define the parallel part before: 
```python
parallel_setup = Parallel(n_jobs=self.n_jobs, backend=self.backend, verbose=self.verbose)
```

Reply:
Will change